SofaKernel/framework/sofa/helper/system/FileRepository.cpp

```diff
@@ -43,6 +43,12 @@
 #include <sstream>
 #include <sofa/helper/logging/Messaging.h>
 
+#ifdef WIN32
+#define ON_WIN32 true
+#else
+#define ON_WIN32 false
+#endif // WIN32
```

Comment:
Yes, as for the NDEBUG stuff we should generalized this kind of design pattern when possible. 
But I suggest to make that in a separate issue in which we try to remove as much as possible of the NDEBUG or the WIN32 focusing only on these aspects;